In [25]:
import sys
import os

sys.path.append(os.path.abspath(".."))

In [26]:
from src.database import get_engine
import pandas as pd

#### Look for null or missing data and take the best option at this case

In [5]:
engine = get_engine()

df = pd.read_sql("SELECT * FROM products", con=engine)
print(df.isnull().sum())

product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product_length_cm               2
product_height_cm               2
product_width_cm                2
dtype: int64


In [24]:
df_items = pd.read_sql("SELECT * FROM order_items", con=engine)

null_products = df.loc[df["product_category_name"].isnull(),
                       'product_id'
                       ].tolist()

df_related = df_items[
    df_items['product_id'].isin(null_products)
]

print(df_related['order_id'].count())

1603


##### We can see that we have 610 products without name and also we have 1603 items related with some of this products, so ourRevenue KPI will be analyzed by orders information.